torch/_refs/__init__.py

```diff
@@ -918,6 +919,47 @@ def trunc(a):
     return prims.trunc(a)
 
 
+# TODO: register this as a real ref/decomposition once TorchInductor supports complex!
+def view_as_complex(self: TensorLikeType) -> TensorLikeType:
+    input_dtype = self.dtype
+    torch._check(
+        utils.is_float_dtype(input_dtype),
+        lambda: f"view_as_complex is only supported for floating point"
+        f"tensors, but got a tensor of scalar type: {input_dtype}",
+    )
+    sizes = self.size()
+    torch._check(
+        len(sizes) != 0,
+        lambda: "Input tensor must have one or more dimensions",
+    )
+    torch._check(
+        sizes[-1] == 2,
+        lambda: "Tensor must have a last dimension of size 2",
+    )
+
+    old_strides = self.stride()
+    new_sizes = sizes[:-1]
+    torch._check(
+        old_strides[-1] == 1,
+        lambda: "Tensor must have a last dimension with stride 1",
+    )
+    dims = old_strides[:-1]
+    torch._check(
+        py_all(stride % 2 == 0 for stride in dims),
+        lambda: "Tensor must have a stride divisible by 2 for all but last dimension",
+    )
+    new_strides = tuple(dim // 2 for dim in dims)
+    torch._check(
+        self.storage_offset() % 2 == 0,
+        lambda: "Tensor must have a storage_offset divisible by 2",
+    )
+    new_storage_offset = self.storage_offset() // 2
+    if not utils.is_complex_dtype(input_dtype):
+        self = prims.view_of_dtype(self, utils.corresponding_complex_dtype(input_dtype))
+
+    return self.as_strided(new_sizes, new_strides, new_storage_offset)
```

Comment:
This decomposition isn't ideal for inductor since it 1) goes directly to prims 2) calls `as_strided`. It would make more sense to decompose into `aten.view.dtype`.

```suggestion
    complex_dtype = utils.corresponding_complex_dtype(input_dtype)
    return self.view(complex_dtype)[..., 0]
```

Which you could in turn implement with `prims.view_of_dtype`.

Reply:
Another reason to do it this way is to fix the meta registration for `view.dtype` which currently isn't a view and doesn't handle different sized dtypes at all!

https://github.com/pytorch/pytorch/blob/808e0886157055a36924afe93cefcce4f1da4c35/torch/_meta_registrations.py#L2645-L2647

Reply:
Alright. I think I will work on the decomposition for `aten.view.dtype` (which means i also need to fix the meta). Will do that in another PR. Thanks for the review!